processing/src/main/java/io/druid/query/lookup/LookupReferencesManager.java

```diff
@@ -91,7 +122,12 @@ public boolean put(String lookupName, final LookupExtractorFactory lookupExtract
       if (!lookupExtractorFactory.start()) {
         throw new ISE("start method returned false for lookup [%s]", lookupName);
       }
-      return (null == lookupMap.putIfAbsent(lookupName, lookupExtractorFactory));
+
+      boolean isAdded = (null == lookupMap.putIfAbsent(lookupName, lookupExtractorFactory));
+      if (isAdded && lookupSnapshotTaker != null) {
+        lookupSnapshotTaker.takeSnapshot(getAllAsList());
```

Comment:
we don't expect this will change very frequently and size will be limited so i think let's keep it this way